build-tools/packages/build-cli/test/library/packageExports.test.ts

```diff
@@ -0,0 +1,331 @@
+		]),
+		commonExportResults: new Map<string, ExportData>([
+			[
+				"Public",
+				{
+					relPath: `./${path}/public.d.ts`,
+					conditions: [condition, "types"],
+					isTypeOnly: false,
+				},
+			],
+			[
+				"Beta",
+				{
+					relPath: `./${path}/beta.d.ts`,
+					conditions: [condition, "types"],
+					isTypeOnly: false,
+				},
+			],
+			// ["Alpha", { relPath: `./${path}/alpha.d.ts`, conditions: [condition, "types"], isTypeOnly: false }],
+			// ["Types", { relPath: `./${path}/types.d.ts`, conditions: [condition, "types"], isTypeOnly: false }],
+		]),
+		commonNode10CompatExportResults: new Map<string, Node10CompatExportData>([
+			[
+				`./beta.d.ts`,
+				{
+					relPath: `./${path}/beta.d.ts`,
+					isTypeOnly: false,
+				},
+			],
+			[
+				`./internal.d.ts`,
+				{
+					relPath: `./${path}/index.d.ts`,
+					isTypeOnly: false,
+				},
+			],
+		]),
+		typeOnlyExportResults: new Map<string, ExportData>([
+			[
+				"Public",
+				{
+					relPath: `./${path}/public.d.ts`,
+					conditions: [condition, "types"],
+					isTypeOnly: true,
+				},
+			],
+			// ["Beta", { relPath: `./${path}/beta.d.ts`, conditions: [condition, "types"], isTypeOnly: true }],
+			[
+				"Alpha",
+				{
+					relPath: `./${path}/alpha.d.ts`,
+					// "./alpha" is setup with a "default" instead of normal "import" condition
+					conditions: condition !== "import" ? ["types", condition] : ["types"],
+					isTypeOnly: true,
+				},
+			],
+			[
+				"Types",
+				{
+					relPath: `./${path}/types.d.ts`,
+					conditions: ["types", condition],
+					isTypeOnly: true,
+				},
+			],
+		]),
+		doubleReferenceExportResults: new Map<string, ExportData>([
+			[
+				"Public",
+				{
+					relPath: `./${path}/public.d.ts`,
+					conditions: [condition, "types"],
+					isTypeOnly: false,
+				},
+			],
+			[
+				"Beta",
+				{
+					relPath: `./${path}/beta.d.ts`,
+					conditions: ["types", condition],
+					isTypeOnly: false,
+				},
+			],
+			// ["Alpha", { relPath: `./${path}/alpha.d.ts`, conditions: [condition, "types"], isTypeOnly: false }],
+			// ["Types", { relPath: `./${path}/types.d.ts`, conditions: [condition, "types"], isTypeOnly: false }],
+		]),
+	};
+}
+
+class MockLogger implements Logger {
+	public calls: [string | Error | undefined, ...any[]][] = [];
+
+	log(message?: string, ...args: any[]): void {
+		this.calls.push([message, ...args]);
+	}
+	info(msg: string | Error | undefined, ...args: any[]): void {
+		this.calls.push([msg, ...args]);
+	}
+	warning(msg: string | Error | undefined, ...args: any[]): void {
+		this.calls.push([msg, ...args]);
+	}
+	errorLog(msg: string | Error | undefined, ...args: any[]): void {
+		this.calls.push([msg, ...args]);
+	}
+	verbose(msg: string | Error | undefined, ...args: any[]): void {
+		this.calls.push([msg, ...args]);
+	}
+}
+
+function assertEquivalentMaps<TKeys, TValues>(
+	actual: Map<TKeys, TValues>,
+	expected: Map<TKeys, TValues>,
+): void {
+	assert.hasAllKeys(actual, [...expected.keys()]);
```

Comment:
`assert.hasAllKeys` confirms that keys are identical - those keys and only those keys (I think the name could be better). The below would allow extra keys.

Reply:
You could simplify all of this as something like 
```suggestion
	assert.deepEquals([...actual.entries()], [...expected.entries()]);
```

I don't know if the error messages would be as useful though.

Reply:
I liked the output of hasAllKeys. And then there is some nicety to being able to break/debug on single value. Thanks though :)